File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String,Integer> mp = new HashMap<>();
        for (String temp : completion){
            if (mp.get(temp) != null){
                mp.put(temp, mp.get(temp) + 1);
            }
            else mp.put(temp, 1);
        }
        for (String s : participant){
            if (mp.get(s) != null){
                if (mp.get(s) == 0){
                    answer = s;
                    break;
                }
                mp.put(s, mp.get(s) - 1);
            }
            else {
                answer = s;
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    map<string,int> mp1,mp2;
    for (string temp : participant){
        mp1[temp]++;
    }
    for (string temp : completion){
        mp2[temp]++;
    }
    for (string temp : participant){
        if (mp1[temp] != mp2[temp]){
            return temp;
        }
    }
    return "";
}